demos/social_distance_demo/cpp/main.cpp

```diff
@@ -214,16 +202,27 @@ class Drawer : public Task {
 // draws results on the frame
 class ResAggregator : public Task {
 public:
-    ResAggregator(const VideoFrame::Ptr &sharedVideoFrame, std::list<cv::Rect> &&boxes,
-                  std::list<TrackableObject> &&trackables)
-            : Task{sharedVideoFrame, 4.0},
-              boxes{std::move(boxes)},
-              trackables{std::move(trackables)} {}
+    ResAggregator(const VideoFrame::Ptr& sharedVideoFrame, std::list<cv::Rect>&& boxes,
+        std::list<TrackableObject>&& trackables)
+        : Task{ sharedVideoFrame, 4.0 },
+        context(static_cast<ReborningVideoFrame*>(sharedVideoFrame.get())->context),
+        sourceID(sharedVideoFrame->sourceID),
+        frameID(sharedVideoFrame->frameId),
+        lastProcessedFrameID(context.trackersContext.lastProcessedIDs[sourceID]),
+        boxes{std::move(boxes)},
+        trackables{std::move(trackables)} {}
+
+    bool isReady() override {
+        return lastProcessedFrameID == frameID;
```

Comment:
They either must be atomic or be wrapped with mutex which is associated with writing to `lastProcessedFrameID` (and `frameID` if it is ever modified, but it think it's not)

Reply:
Added mutex lock.

Reply:
Made `lastProcessedFramedId` as `std::atomic<int64_t>` type.